webapp/packages/plugin-data-viewer/src/DatabaseDataModel/Actions/ResultSet/ResultSetDataAction.ts

```diff
@@ -94,6 +99,14 @@ export class ResultSetDataAction extends DatabaseDataResultAction<IResultSetElem
     return this.rows[row.index];
   }
 
+  getRowMetadata(row: IResultSetRowKey) {
+    if (row.index >= this.rowsWithMetadata.length) {
+      return undefined;
+    }
+
+    return this.rowsWithMetadata[row.index].metaData;
```

Comment:
you can remove `if` above and use this:

`return this.rowsWithMetadata?.[row.index].metaData;` - to also include case where something went wrong  on backend side, and you received a negative value index

Reply:
you will get mobx warnings when index bigger or less then array length